xapian-core/include/xapian/cluster.h

```diff
@@ -28,14 +28,45 @@
 #endif
 
 #include <xapian/mset.h>
+#include <xapian/queryparser.h>
 #include <xapian/types.h>
 #include <xapian/visibility.h>
 
 #include <unordered_map>
+#include <unordered_set>
 #include <vector>
 
 namespace Xapian {
 
+/// Stopper subclass which checks for both stemmed and unstemmed stopwords
+class XAPIAN_VISIBILITY_DEFAULT StemStopper : public Xapian::Stopper {
+  public:
+    /// Stemming strategies
+    typedef enum { STEM_NONE, STEM_SOME, STEM_ALL_Z } stem_strategy;
```

Comment:
Currently, the main idea being use to check whether a term is stemmed or not is checking it's first term. If `term[0] == 'Z'`, we allow the term to be added in if it's not a stopword. STEM_NONE and STEM_ALL seem inappropriate here because they were the reason the tests were failing before. If we do STEM_ALL, that would store stemmed terms without the 'Z' prefix which would lead to an empty termlist. In this case, do we need to change the way we look for stemmed terms?